lib/internal/http2/core.js

```diff
@@ -441,6 +441,8 @@ function afterDoStreamWrite(status, handle, req) {
 
 function onHandleFinish() {
   var session = this[kSession];
+  if (!session)
```

Comment:
Btw this feels like a code smell. I found that `onHandleFinish` always runs twice. Second time the session is `undefined`, hence the guard.